firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/crcelements.txt

```diff
@@ -0,0 +1,186 @@
+      THEN("The output buffer is as expected")
+
+  GIVEN("A crc element constructed with an empty payload")
+
+    WHEN("The crc and payload are written to the output buffer")
+      THEN("the crc accessor method returns 0 before the write method is called")
+      THEN("After the write method is called, the crc accessor method returns 0")      
+      THEN("The CRC field of the body's header matches the value returned by the crc accessor method.")
+      THEN("The body's payload section is empty")
+      THEN("The output buffer is as expected")
+
+/* parse function
+Scenario: CRCElement: The parse method correctly updates its internal crc value and the constructor's payload buffer from the input buffer
+  
+  GIVEN("A CRCElement constructed with an empty payload buffer with a capacity of 254 bytes")
+
+    WHEN("A body without a complete 4-byte header is parsed")
+      THEN("the parse method reports out of bounds status")
+      THEN("the crc accessor method returns 0")
+      THEN("the payload accessor method returns a reference to the payload buffer given in the CRCElement constructor")
+      THEN("the payload given in the CRCElement constructor is independent of the input buffer body")
+
+    WHEN("A body with a complete 4-byte header, and a payload consistent with the CRC field of the header, is parsed")
+      THEN("the parse method reports ok status")
+      THEN("After the parse method is called, the value returned by the crc accessor method is equal to the crc field of the input body's header")
+      THEN("the payload returned from the payload accessor method is equal to the payload from the body")
+      THEN("the payload accessor method returns a reference to the payload buffer given in the CRCElement constructor") 
+      THEN("the payload given in the CRCElement constructor is independent of the input buffer body")
+
+    WHEN("A body with a complete 4-byte header, and a payload inconsistent with the CRC field of the header, is parsed")
+      THEN("the parse method reports ok status")
+      THEN("After the parse method is called, the value returned by the crc accessor method is equal to the crc field of the input body's header")
+      THEN("the value returned by the crc accessor method is inconsistent with the payload")
+      THEN("the payload returned from the payload accessor method is equal to the payload from the body")
+      THEN("the payload accessor method returns a reference to the payload buffer given in the CRCElement constructor")
+      THEN("the payload given in the CRCElement constructor is independent of the input buffer body")
+
+  GIVEN("A CRCElement constructed with a non-empty payload buffer of capacity 254 bytes")
+
+    WHEN("A body without a complete 4-byte header is parsed")
+      THEN("the parse method reports out of bounds status")
+      THEN("the value returned by the crc accessor method is equal to the crc field of the input body's header")
```

Comment:
Actually, the CRC remains at whatever value it was given before parse was called.